src/serialize.cpp

```diff
@@ -37,15 +38,64 @@ void raw_data_to_value(value& v, const RawData& rd)
     if(rd.get_shape().type() == shape::tuple_type)
         result["sub"] = migraphx::to_value(rd.get_sub_objects());
     else if(not rd.empty())
-        result["data"] = migraphx::value::binary(rd.data(), rd.get_shape().bytes());
+    {
+        size_t binary_size      = rd.get_shape().bytes();
+        size_t partition_length = std::numeric_limits<uint32_t>::max();
+        if(binary_size > partition_length)
+        {
+            size_t array_size = 1 + ((binary_size - 1) / partition_length);
+            std::vector<migraphx::value> v_array(array_size);
+            for(size_t i = 0; i < array_size; ++i)
+            {
+                size_t chunk_size =
+                    (i == (array_size - 1)) ? (binary_size % partition_length) : partition_length;
+                v_array[i] =
+                    migraphx::value::binary{(rd.data() + (i * partition_length)), chunk_size};
+            }
```

Comment:
This should be done in msgpack.cpp. This is a limitation of msgpack not of the value class.

Reply:
Inside msgpack while unpacking, i do not have a way to determine if array<binary> is for the literal or is it really just array<binary> . If it is for the literal, i want to bundle array<binary> as single value::binary.

Reply:
We should do this chunking for all arrays and objects.

Reply:
Here's my attempt to do chunking for msgpack only: https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2059